crypto3/libs/algebra/include/nil/crypto3/algebra/fields/detail/element/fp12_2over3over2.hpp

```diff
@@ -143,11 +143,6 @@ namespace nil {
                             return *this;
                         }
 
-                        element_fp12_2over3over2 sqrt() const {
```

Comment:
I see we have sqrt for fp2. Doesn't that algorithm apply here? Are we sure we don't need sqrt for this extension fields?

Reply:
We need sqrt only for fields that can be used as base fields for curves.
For pairing-friendly curves these are:
Fp2 for MNT4::G2
Fp3 for MNT6::G2
Fp2 for BLS12::G2

Sqrt is specifically used for marshalling a compressed form of a point - we store only X (or Y) coordinate, then solve the equation, like this: $y^2= x^3+bx+c$ to recover Y coordinate.

Fields for GT groups - Fp4 (GT for MNT4), Fp6 (GT for MNT6)  and Fp12  (GT for BLS12) do not serve as base fields for elliptic curve elements and thus do not require sqrt function. In most places it was empty with TODO.